### ISSUE TITLE
[$500] When creating a group, members list does not show the account which created the group

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#33624]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.17-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1703532852814079

## Action Performed:
1. Open app and login as account A
2. Click fab button > start chat
3. Add couple more members to the group
4. Click on the header and check the members list

## Expected Result:
Members list should include the account A as well

## Actual Result:
Not includes account A

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Snip - (27) New Expensify - Google Chrome (2)](https://github.com/Expensify/App/assets/38435837/599e47af-1d9a-45e4-8d3a-95242c8afac9)
![Snip - (27) New Expensify - Google Chrome](https://github.com/Expensify/App/assets/38435837/a45b9d8e-e5a6-458b-910f-5586cf4322fa)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013e85b3bd39d796fc</li>
        <li>Upwork Job ID: 1739745860553265152</li>
        <li>Last Price Increase: 2023-12-26</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 28076848</li>
<li>FitseTLT | Contributor | 28076849</li>
</ul></li>
    </ul>
</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When creating a group, members list does not show the account which created the group

### What is the root cause of that problem?
We are currently dispaying `participantID` of the the group and owner is not included
https://github.com/Expensify/App/blob/595bf4075988c0b64cb0db1e37694284637fbea2/src/pages/ReportDetailsPage.js#L76-L77
### What changes do you think we should make in order to solve the problem?

We should include `session.accountID`
### What alternative solutions did you explore? (Optional)

**Comment by aim-salam** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When creating a group, members list does not show the account which created the group

### What is the root cause of that problem?

During creation of the group, account doesn't included in the data.

### What changes do you think we should make in order to solve the problem?

During creation of the group, include account data.

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?

We use the `ReportUtils.getParticipantsIDs` function to get the member list. However, the function only takes the `report.participantAccountIDs` and it does not include the creator of the group. Only when `isMoneyRequestReport` is true is when the `report.managerID` and `report.ownerAccountID` are added.

### What changes do you think we should make in order to solve the problem?


Add the id of the account creator into the array return by `ReportUtils.getParticipantsIDs` in `ReportParticipantsPage`.

**Comment by KrAbhas** [comment]
## Proposal

Please re-state the problem that we are trying to solve in this issue

The user who created the account is not added to the list of members in a chat room.

What is the root cause of that problem?

It gets root all the members that are added to the report in onyx. Based on that it just iterates over the data and gets the members.
`const participants = _.map(getAllParticipants(props.report, props.personalDetails, props.translate), (participant) => ({
        ...participant,
        isDisabled: ReportUtils.isOptimisticPersonalDetail(participant.accountID),
    }));`

What changes do you think we should make in order to solve the problem?

Add the current user as well to the chatroom while creating chat in `createGroup` we can add the current user id for participant list.

What alternative solutions did you explore? (Optional)
NA

**Comment by dragnoir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When creating a chat group, members list does not show the user who created the group

### What is the root cause of that problem?

 `report.participantAccountIDs`  is missing the account user for chatReport

### What changes do you think we should make in order to solve the problem?

Update  `getParticipantsIDs`  to include `currentUserAccountID` when `isChatReport` is true

```js
function  getParticipantsIDs(report:  OnyxEntry<Report>):  number[] {
// ... same code
	if (isChatReport(report) &&  !isChatRoom(report) &&  !isPolicyExpenseChat(report) &&  !isChatThread(report) &&  !isTaskReport(report)) {

		const  chatReportParticipants  = [currentUserAccountID, ...participants].filter(Boolean) as  number[];
		return  chatReportParticipants;
	}

	return  participants;
}
```

**Comment by dragnoir** [comment]
`isGroupChat(report)` will not work is case of a chat between 2 participants and the chat creator.

https://github.com/Expensify/App/blob/9e81447833a4b9f9ea34a91f01185b0687ddb66a/src/libs/ReportUtils.ts#L4121-L4131

`(report.participantAccountIDs?.length ?? 0) > 2` require a total of 3 participants at least. (This may be a bug)

**Comment by bfitzexpensify** [comment]
Few proposals ready for review @akinwale - thank you!

**Comment by akinwale** [comment]
This is a fairly straightforward issue, so we can go with @FitseTLT's [proposal](https://github.com/Expensify/App/issues/33624#issuecomment-1869762034) since they were first to find the root cause and suggest a working solution.

🎀👀🎀 C+ reviewed.

**Comment by MonilBhavsar** [comment]
@dragnoir group chat requires at least 3 participants. For two participants it is simple DM chat report.

**Comment by MonilBhavsar** [comment]
@FitseTLT in your alternate solution why do you think we need condition other than isGroupChat()?

**Comment by dragnoir** [comment]
@MonilBhavsar isGroupChat require >2, equal to min 3 participants.
Those 3 participants required by isGroupChat does not include the user who created the group chat. When we add this user (chat creator), the total is now equal to 4. That means any group with the creator + 2 participants in not considered true by isGroupChat.

**Comment by FitseTLT** [comment]
> @FitseTLT in your alternate solution why do you think we need condition other than isGroupChat()?

@MonilBhavsar It's because participants array only includes memebers of the group chat without the current user and `isGroupChat` requires `participants.length > 2` so if the group chat has 3 members it is `participant.length = 2` and our fix will not be applied. So if there is no special reason to stick to it we should either update `isGroupChat` accordingly or only use the updated logic for our specific case here.

**Comment by MonilBhavsar** [comment]
Thank you both for clarification and proposals!
Since @FitseTLT figured it out first, we'll move forward with their proposal

**Comment by FitseTLT** [comment]
So @MonilBhavsar what's your opinion on `isGroupChat`
1. Should I use `isGroupChat` with modifying the logic inside it
2. use modified logic for this fix only
3. Update `isGroupChat` logic from `participants.length > 2` to `participants.length >=2`

**Comment by MonilBhavsar** [comment]
`isGroupChat()` logic is wrong, no?
So we should fix that for participants greater than > 1?

**Comment by akinwale** [comment]
> `isGroupChat()` logic is wrong, no? So we should fix that for participants greater than > 1?

We used >2 to differentiate from 1:1 DMs. [Context](https://github.com/Expensify/App/pull/28991#discussion_r1350694994).

The proposed approach of adding the group creator to the list of participant account IDs when checking makes sense in this scenario.

**Comment by Tony-MK** [comment]
@akinwale, since the bug also occurs in the [ReportDetailsPage](https://github.com/Expensify/App/blob/2b58e1cbef63c4fdce8a6e8fb3ee52089009a8d8/src/pages/ReportDetailsPage.js#L76) shouldn't we also update it.

Is it unnecessary?

**Comment by MonilBhavsar** [comment]
> We used >2 to differentiate from 1:1 DMs. [Context](https://github.com/Expensify/App/pull/28991#discussion_r1350694994).
> 
> The proposed approach of adding the group creator to the list of participant account IDs when checking makes sense in this scenario.

Thanks for the context 👍 
I agree. @FitseTLT hope this comment is resolved

**Comment by MonilBhavsar** [comment]
> since the bug also occurs in the [ReportDetailsPage](https://github.com/Expensify/App/blob/2b58e1cbef63c4fdce8a6e8fb3ee52089009a8d8/src/pages/ReportDetailsPage.js#L76) shouldn't we also update it.

We're updating `getParticipantsIDs()` so it should fix for all the pages

**Comment by FitseTLT** [comment]
> > We used >2 to differentiate from 1:1 DMs. [Context](https://github.com/Expensify/App/pull/28991#discussion_r1350694994).
> > The proposed approach of adding the group creator to the list of participant account IDs when checking makes sense in this scenario.
> 
> Thanks for the context 👍 I agree. @FitseTLT hope this comment is resolved

I assume you guys are agreeing using a modified logic than `isGroupChat` But shall I use it only for this fix or U expect to change `isGroupChat` which obviously affect other part of the codebase too???

**Comment by MonilBhavsar** [comment]
Sorry didn't get your last message. Are you suggesting other refactor than mentioned in your proposal here https://github.com/Expensify/App/issues/33624#issuecomment-1869762034?

**Comment by FitseTLT** [comment]
@MonilBhavsar take a look at my previous [comment](https://github.com/Expensify/App/issues/33624#issuecomment-1875793460)

**Comment by MonilBhavsar** [comment]
I would vote for updating `isGroupChat()` and evaluating that we don't break existing 3 usages

**Comment by FitseTLT** [comment]
Agree. I will make sure!!

**Comment by dragnoir** [comment]
@MonilBhavsar I am the 1st who mentioned fixing `isGroupChat()` in my alternative solution!!!

**Comment by bfitzexpensify** [comment]
Payments complete, reminder on the BZ checklist @akinwale

**Comment by akinwale** [comment]
> Payments complete, reminder on the BZ checklist @akinwale

Sorry, I missed the previous notification. The PR for this issue was reverted, with a new one currently on hold waiting for feedback from @MonilBhavsar.

**Comment by MonilBhavsar** [comment]
Removed the payment label and adjusted title. I'll take a look today

**Comment by MonilBhavsar** [comment]
@marcaaron I would like to get your thoughts on this issue since it coincides with the Group chats project.
The condition here is responsible for not returning the current participant in the list of participants and hence we do not display the current participant in members list 
https://github.com/Expensify/Auth/blob/261d3458ed9673843dec55ece1cd1466f847af35/auth/lib/Report.cpp#L5773-L5776

The `chatType` is empty currently for group chats created from newDot. 

I am thinking to fix it by not skipping it is a group chat i.e.`type=chat && participants.size() > 2`

I also know that group chats is in parallel and briefly looking at the doc, we're restructuring the participants and also include  a role for each of them. What's your take on me going ahead and fixing this issue by updating the condition linked above, or
do nothing for this bug as it will be resolved with Group chats refactor.

**Comment by marcaaron** [comment]
>What's your take on me going ahead and fixing this issue by updating the condition linked above, or
do nothing for this bug as it will be resolved with Group chats refactor.

It sounds OK, but maybe not the best use of time if this isn't something that we want fixed urgently for some reason. There is some overlap where old Group DMs will exist at the same time as new Group Chats before we migrate them over completely. But ultimately whatever change you make to fix this might be rolled back in the near future.

**Comment by MonilBhavsar** [comment]
Thanks for the thoughts!
At this point of time it sounds good to do it as a part of Group chats project.

**Comment by bfitzexpensify** [comment]
Cool, let's close this out for the moment then.
